### PR TITLE
Add image replacement support for multicontainer pods in kubectl rolling-update

### PR DESCRIPTION
**Pull request by mhenniges**
Added support to kubectl rolling-update with the --image flag when operating on multi-container pods. The container to update is selected by the new --container-name option.

All existing behavior of kubectl was preserved.

Thanks for reading!

### CODE REVIEW

**Comment by mhenniges** [comment]
Too many github profiles...commits were logged with wrong account. Will try to fix the CLA setup.

**Comment by erictune** [comment]
`Boilerplate header is wrong for: ./pkg/kubectl/rolling_updater.go`

Make sure you did:
`cd .git/hooks; ln -s ../../hooks/pre-commit .`
as described in `docs/devel/development.md`

**Comment by erictune** [comment]
Your commits are authored by matthewsimon-ww who has not signed the CLA.
The PR is from mhenniges who has signed the CLA.

**Comment by erictune** [comment]
Looks like this fixes #9868

**Comment by erictune** [comment]
Assigned to davidopp who has previously shown interested in rolling updated.  Feel free to punt elsewhere.

**Comment by mhenniges** [comment]
Hi, you are mistaken, they were authored by me.

Thanks.

**Comment by mhenniges** [comment]
Thanks for the feedback erictune, I've resolved the boilerplate and other issues as per the hooks, and also fixed a unit test I missed earlier.

**Comment by davidopp** [comment]
Thanks very much for the PR!

Unfortunately, we're currently in "code freeze" in preparation for our 1.0 release, so we can't merge this right now. However, this feature is obviously useful and we can definitely merge it after the code freeze is over.

In the mean time, it would be great if you could add some tests.

File: pkg/kubectl/cmd/rollingupdate.go [hunk 1]
```diff
@@ -74,6 +74,7 @@ func NewCmdRollingUpdate(f *cmdutil.Factory, out io.Writer) *cobra.Command {
 	cmd.Flags().String("timeout", timeout, `Max time to wait for a controller to update before giving up. Valid time units are "ns", "us" (or "µs"), "ms", "s", "m", "h".`)
 	cmd.Flags().StringP("filename", "f", "", "Filename or URL to file to use to create the new controller.")
 	cmd.Flags().String("image", "", "Image to upgrade the controller to.  Can not be used with --filename/-f")
+	cmd.Flags().String("container-name", "", "Container name which will have its image upgraded. Only relevant when --image is specified, ignored otherwise. Required when using --image on a multi-container pod")
```
Comment (bgrant0607) [inline pkg/kubectl/cmd/rollingupdate.go]: Please conform to the convention of other commands, such as log and exec: `--container`, `-c`
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/kubectl/cmd/log.go#L85
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/kubectl/cmd/exec.go#L68

**Comment by bgrant0607** [comment]
We're trying to clear the PR backlog. Please rebase.

cc @nikhiljindal

**Comment by mhenniges** [comment]
What branch should I rebase to? Master? Or the 1.0?

**Comment by jlowdermilk** [comment]
Rebase to master. I believe release branches are considered final except
for p0 bugfixes/security patches.

**Comment by bgrant0607** [comment]
Yes, master, sorry.

**Comment by bgrant0607** [comment]
cc @kubernetes/kubectl

File: docs/kubectl_rolling-update.md [hunk 2]
```diff
@@ -36,6 +36,7 @@ $ kubectl rolling-update frontend --image=image:v2
 ### Options
 
 ```
+      --container-name="": Container name which will have its image upgraded. Only relevant when --image is specified, ignored otherwise. Required when using --image on a multi-container pod
```
Comment (bgrant0607) [inline docs/kubectl_rolling-update.md]: Please be consistent with other commands that have container-name flags. For instance:
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/exec.go#L68

**Comment by bgrant0607** [comment]
Please rebase.

**Comment by JanetKuo** [comment]
Ping @mhenniges could you rebase?

@k8s-bot ok to test

**Comment by bgrant0607** [comment]
This PR appears to be inactive since July. Closing.

**Comment by coleca** [comment]
So this will never be added to K8S?

**Comment by JanetKuo** [comment]
@coleca please see https://github.com/kubernetes/kubernetes/issues/13483#issuecomment-149979546

**Pull request closed without merging**